diff --git a/NAM/get_dsp.cpp b/NAM/get_dsp.cpp
index 6439364..f62d07e 100644
--- a/NAM/get_dsp.cpp
+++ b/NAM/get_dsp.cpp
@@ -1,7 +1,6 @@
 #include <fstream>
 #include <iostream>
 #include <mutex>
-#include <regex>
 #include <sstream>
 #include <stdexcept>
 
@@ -11,6 +10,8 @@
 #include "get_dsp.h"
 #include "model_config.h"
 
+#include <ctre/ctre.hpp>
+
 namespace nam
 {
 namespace
@@ -21,8 +22,8 @@ class CoreVersionSupportChecker : public IVersionSupportChecker
 public:
   Supported support(const std::string& version) const override
   {
-    static const std::regex semver_regex(R"(^\d+\.\d+\.\d+$)");
-    if (!std::regex_match(version, semver_regex))
+    static constexpr auto semver_regex = ctre::match<R"(^\d+\.\d+\.\d+$)">;
+    if (!semver_regex(version))
       return Supported::NO;
 
     const Version parsed = ParseVersion(version);
